packages/rule-typescript-config/src/target.ts

```diff
@@ -0,0 +1,235 @@
+
+
+                if (!previousVersion || isOlder(version, previousVersion)) {
+                    config[browser] = version;
+                }
+
+                return config;
+            }, {});
+
+            return configuration;
+        };
+
+        /**
+         * Based on the minimum supported browser configuration passed,
+         * determines what's the maximum ES version that can be targetted.
+         */
+        const getMaxVersion = (minimumBrowsers): string => {
+            const versions = Object.keys(compatMatrix);
+            let maxVersion = 'es3';
+
+            /**
+             * This will check all the ES versions and compare all the
+             * browsers from the compat matrix agains the given ones.
+             * If there's one in the compat matrix newer than the ones
+             * provided, it will fail.
+             */
+            versions.forEach((version) => {
+                /** The list of browsers to take into account for the compat matrix. */
+                const browsers: [string, string | number][] = Object.entries(compatMatrix[version]);
+
+                const validates = browsers.reduce((valid, [browser, minimumBrowserVersion]) => {
+                    const minimumTargettedBrowserVersion = minimumBrowsers[browser];
+
+                    // The user doesn't care about this browser, there isn't a version
+                    if (!minimumTargettedBrowserVersion) {
+                        return valid;
+                    }
+
+                    /*
+                     * The user cares about this browsers but there's no browsers version that supports
+                     * this ES release.
+                     * E.g.: ES2015 and IE11
+                     */
+                    if (minimumBrowserVersion === 'none' && minimumTargettedBrowserVersion) {
+                        return false;
+                    }
+
+                    /**
+                     *  `isOlder` checks if it's strictly older. We do
+                     * `!isOlder(minimumTargettedBrowserVersion, minimumBrowserVersion)` instead of
+                     * `isOlder(minimumBrowserVersion, minimumTargettedBrowserVersion)`
+                     * just in case they are the same version
+                     */
+                    const newer = !isOlder(minimumTargettedBrowserVersion, minimumBrowserVersion);
+
+                    return valid && newer;
+                }, true);
+
+                maxVersion = validates ? version : maxVersion;
+            });
+
+            return maxVersion;
+        };
+
+        /**
+         * Returns the configured normalized target:
+         *
+         * * The `es` part will be upper cased (e.g.: ES2015, ESNext)
+         * * ES6 --> ES2015
+         */
+        const getConfiguredTarget = (config: TypeScriptConfig): string => {
+            const target = config && config.compilerOptions && config.compilerOptions.target || 'es3';
```

Comment:
I think you don't need to check if `config` exists, if you receive an `parse::typescript-config` then the config has to exist.